src/pricing-helper.ts

```diff
@@ -172,6 +172,20 @@ export class PricingHelper {
           return false;
         }
 
+        if (Array.isArray(p.gasCost)) {
+          if (p.gasCost.length !== amounts.length) {
+            this.logger.error(
+              `Error_getPoolPrices: ${p.exchange} returned prices with invalid gasCost array length: ${p.gasCost.length} !== ${amounts.length}`,
+            );
+            return false;
+          } else if (amounts[0] === 0n && p.gasCost[0] === 0) {
+            this.logger.error(
```

Comment:
This condition doesn't seem correct

Reply:
I think it should be
`if(amounts[0] === 0n && p.gasCost[0] !== 0)`

but may be its safer to check for every amount which is 0 if we don't want to make assumptions on the amounts

Reply:
Fixed